lib/smb2/file.rb

```diff
@@ -34,47 +62,53 @@ def initialize(tree:, create_response:)
   #
   # @return [Smb2::Packet::CloseResponse]
   def close
-    packet = Smb2::Packet::CloseRequest.new do |request|
-      request.file_id = self.create_response.file_id
-    end
+    packet = Smb2::Packet::CloseRequest.new(
+      file_id: self.create_response.file_id
+    )
 
     response = tree.send_recv(packet)
 
     Smb2::Packet::CloseResponse.new(response)
   end
 
+  # Whether the current read/write pointer is at the end of the file
+  #
+  # @return [Boolean]
+  def eof?
+    (last_read_response && last_read_response.header.nt_status == STATUS_END_OF_FILE) || pos == size
+  end
+
   # @return [String]
   def inspect
-    "#<Smb2::File file-id=#{create_response.file_id.unpack("H*").first} >"
+    "#<Smb2::File:#{tree.share}\\#{filename} file-id=#{create_response.file_id.unpack("H*").first} >"
   end
 
-  # Call {#read_chunk} repeatedly until we get everything
+  # Call {#read_chunk} repeatedly until we get `length` bytes or hit {#eof?
+  # end of file}.
   #
   # @note Beware of ballooning memory usage
   # @note Calling this on a pipe without a `length` is probably a really bad
   #   idea
   #
-  # @param offset [Fixnum] offset from the beginning of the file (*default*: 0)
   # @param length [Fixnum,nil] number of bytes to read. If this is nil, read
   #   the whole file, i.e. {#size} bytes.
   # @return [String] full contents of the remote file
-  def read(offset: 0, length: nil)
+  def read(length = nil)
```

Comment:
This change is causing a spec failure due to differences in method arity between here and what the spec expects to use...